pkg/resource/fake/mocks.go

```diff
@@ -225,67 +189,14 @@ func (o *Object) DeepCopyObject() runtime.Object {
 	return out
 }
 
-// Claim is a mock that implements Claim interface.
-type Claim struct {
-	metav1.ObjectMeta
-	ClassSelector
-	ClassReferencer
-	ManagedResourceReferencer
-	LocalConnectionSecretWriterTo
-	v1alpha1.ConditionedStatus
-	v1alpha1.BindingStatus
-}
-
-// GetObjectKind returns schema.ObjectKind.
-func (m *Claim) GetObjectKind() schema.ObjectKind {
-	return schema.EmptyObjectKind
-}
-
-// DeepCopyObject returns a copy of the object as runtime.Object
-func (m *Claim) DeepCopyObject() runtime.Object {
-	out := &Claim{}
-	j, err := json.Marshal(m)
-	if err != nil {
-		panic(err)
-	}
-	_ = json.Unmarshal(j, out)
-	return out
-}
-
-// Class is a mock that implements Class interface.
-type Class struct {
-	metav1.ObjectMeta
-	Reclaimer
-}
-
-// GetObjectKind returns schema.ObjectKind.
-func (m *Class) GetObjectKind() schema.ObjectKind {
-	return schema.EmptyObjectKind
-}
-
-// DeepCopyObject returns a copy of the object as runtime.Object
-func (m *Class) DeepCopyObject() runtime.Object {
-	out := &Class{}
-	j, err := json.Marshal(m)
-	if err != nil {
-		panic(err)
-	}
-	_ = json.Unmarshal(j, out)
-	return out
-}
-
 // Managed is a mock that implements Managed interface.
 type Managed struct {
 	metav1.ObjectMeta
-	ClassReferencer
-	ClaimReferencer
 	ProviderReferencer
 	ProviderConfigReferencer
 	ConnectionSecretWriterTo
 	Orphanable
-	Reclaimer
```

Comment:
Others are meaningful only if the managed resource was used with a claim, which is being removed. But `Reclaimer` actually corresponds to a behavior in managed resource level. So, that would be a breaking change on `beta` level. In the release notes, we should make sure that this specific change is emphasized.

Reply:
Yeah, I added the `very-breaking-change` to help remind us to pay close attention to this PR when writing release notes.